x/ccv/provider/keeper/keeper_test.go

```diff
@@ -183,30 +174,47 @@ func TestInitHeight(t *testing.T) {
 	}
 }
 
-func TestIterateOverUnbondingOpIndex(t *testing.T) {
+// TestGetAllUnbondingOpIndexes tests GetAllUnbondingOpIndexes behavior correctness
+func TestGetAllUnbondingOpIndexes(t *testing.T) {
 
-	providerKeeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	pk, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
 	defer ctrl.Finish()
 
-	chainID := "6"
+	chainID := "chain-1"
 
-	// mock an unbonding index
-	unbondingOpIndex := []uint64{0, 1, 2, 3, 4, 5, 6}
+	cases := []types.VscUnbondingOps{
+		{
+			VscId:          2,
+			UnbondingOpIds: []uint64{1, 2, 3, 4, 5, 6},
+		},
+		{
+			VscId:          1,
+			UnbondingOpIds: []uint64{1, 2, 3},
+		},
+	}
 
-	// set ubd ops by varying vsc ids and index slices
-	for i := 1; i < len(unbondingOpIndex); i++ {
-		providerKeeper.SetUnbondingOpIndex(ctx, chainID, uint64(i), unbondingOpIndex[:i])
+	pk.SetUnbondingOpIndex(ctx, "chain-2", 1, []uint64{1, 2, 3})
+	for _, c := range cases {
+		pk.SetUnbondingOpIndex(ctx, chainID, c.VscId, c.UnbondingOpIds)
 	}
 
-	// check iterator returns expected entries
-	i := 1
-	providerKeeper.IterateOverUnbondingOpIndex(ctx, chainID, func(vscID uint64, ubdIndex []uint64) (stop bool) {
-		require.Equal(t, uint64(i), vscID)
-		require.EqualValues(t, unbondingOpIndex[:i], ubdIndex)
-		i++
-		return false // do not stop the iteration
-	})
-	require.Equal(t, len(unbondingOpIndex), i)
+	// iterate and check all results are returned
+	result := pk.GetAllUnbondingOpIndexes(ctx, chainID)
+	require.Len(t, result, 2, "wrong result len - should be 2, got %d", len(result))
+	require.Contains(t, result, cases[0], "result does not contain '%s'", cases[0])
+	require.Contains(t, result, cases[1], "result does not contain '%s'", cases[1])
+
+	result = []types.VscUnbondingOps{}
```

Comment:
It seems like the following code could be replaced with 3 lines

```go
result = GetAllUnbondingOpIndexes()
// require len == 1
// require res[0] == expected
```
I see this pattern throughout the file so might be worth changing elsewhere